pcap/pcap.go

```diff
@@ -336,8 +379,27 @@ func (p *Handle) getNextBufPtrLocked(ci *gopacket.CaptureInfo) error {
 	}
 	var result NextError
 	for {
+		// test if we need to stop
+		select {
+		case <-p.stop:
+			return io.EOF
+		default:
+		}
+
+		if p.timeout < 0 {
+			C.pcap_wait(p.cptr, 0)
+		} else {
+			// need to wait less than the read timeout according to pcap
```

Comment:
I interpret that as meaning that it won't work on FreeBSD 4.3-4.6.

Another thing to note in the package documentation? Is it worth trying to test?